src/orquestra/sdk/_base/_logs/_markers.py

```diff
@@ -0,0 +1,136 @@
+
+from .. import _dates
+
+ORQ_MARKER_PREFIX = "ORQ-MARKER:"
+ORQ_MARKER_PATTERN = re.compile(re.escape(ORQ_MARKER_PREFIX) + r"(.+)")
+
+
+# `TaskStartMarker` and `TaskEndMarker` are record types that hold the information we
+# emit in the log markers. They're dataclasses and not pydantic models because we're
+# not just using plain JSON: we're using prefix markers.
+
+
+@dataclass(frozen=True)
+class TaskStartMarker:
+    event = "task_start"
+    wf_run_id: WorkflowRunId
+    task_inv_id: TaskInvocationId
+    timestamp: _dates.Instant
+
+    @property
+    def line(self) -> str:
+        event = {
+            "event": "task_start",
+            "timestamp": _dates.local_isoformat(self.timestamp),
+            "wf_run_id": self.wf_run_id,
+            "task_inv_id": self.task_inv_id,
+        }
+        return f"{ORQ_MARKER_PREFIX}{json.dumps(event)}"
+
+
+@dataclass(frozen=True)
+class TaskEndMarker:
+    event = "task_end"
+    wf_run_id: t.Optional[WorkflowRunId]
+    task_inv_id: t.Optional[TaskInvocationId]
+    timestamp: _dates.Instant
+
+    @property
+    def line(self) -> str:
+        event = {
+            "event": "task_end",
+            "timestamp": _dates.local_isoformat(self.timestamp),
+            **({"wf_run_id": self.wf_run_id} if self.wf_run_id else {}),
+            **({"task_inv_id": self.task_inv_id} if self.task_inv_id else {}),
+        }
+        return f"{ORQ_MARKER_PREFIX}{json.dumps(event)}"
+
+
+Marker = t.Union[TaskStartMarker, TaskEndMarker]
+
+
+def parse_line(line: str) -> t.Optional[Marker]:
+    if not line.startswith(ORQ_MARKER_PREFIX):
+        return None
+
+    if (match := ORQ_MARKER_PATTERN.match(line)) is None:
+        return None
+
+    event = json.loads(match.group(1))
+    try:
+        if event["event"] == TaskStartMarker.event:
+            return TaskStartMarker(
+                wf_run_id=event["wf_run_id"],
+                task_inv_id=event["task_inv_id"],
+                timestamp=_dates.from_isoformat(event["timestamp"]),
+            )
+        elif event["event"] == TaskEndMarker.event:
+            return TaskEndMarker(
+                wf_run_id=event.get("wf_run_id"),
+                task_inv_id=event.get("task_inv_id"),
+                timestamp=_dates.from_isoformat(event["timestamp"]),
+            )
+        else:
+            return None
+    except KeyError:
+        return None
+
+
+def print_start(wf_run_id: WorkflowRunId, task_inv_id: TaskInvocationId):
+    """
+    Emits "task start" marker to stdout and stderr. Required for task-log correlation.
+    """
+    now = _dates.now()
+    marker = TaskStartMarker(
+        wf_run_id=wf_run_id, task_inv_id=task_inv_id, timestamp=now
+    )
+    print(marker.line)
+    print(marker.line, file=sys.stderr)
+
+
+def print_end(wf_run_id: WorkflowRunId, task_inv_id: TaskInvocationId):
+    """
+    Emits "task end" marker to stdout and stderr. Required for task-log correlation.
```

Comment:
Praise: including the intended usage in the docstring!

Suggestion: I think this should probably be a multi-line docstring, line 1 describes the behaviour, line 2 adds context:
```suggestion
    Emits "task end" marker to stdout and stderr. 
    
    Required for task-log correlation.
```